sampling/batch_creator.py

```diff
@@ -8,8 +8,7 @@ def create_batches(self):
     """
       Keep creating batches in the background
     """
-    # Read the samples on memory (data_store.samples) and
-    # generate batches.
+    # Read the samples on memory (data_store.samples) and generate batches.
     # Place the batches in data_store.batches.
-    # Pause if data_store.batches is full.
+    # Pause if data_store.batches is full (or if data_store.samples empty)
```

Comment:
Could you explain your reasoning behind data_store.samples being empty?

Reply:
I think this has to do with our staleness modeling, i.e. how long does our sample persist before we need to get a new one? I assumed that the BatchCreator will remove items from the samples the same way get_next_batch removes a batch from the batches.

Reply:
So let's say we put the logic of declaring a sample "Stale" in the batch_creator. But your comment confuses me, you _pause_ if data_store.samples is empty? Shouldn't the process be creating samples in that case?

Reply:
> So let's say we put the logic of declaring a sample "Stale" in the batch_creator. But your comment confuses me, you _pause_ if data_store.samples is empty? Shouldn't the process be creating samples in that case?

As we discussed, both would be independent. The batch creator just gets the samples if there are samples available. Sample Creator keeps creating the samples in the background, it stops only if the process gets killed. Let me know if this is not correct.

Reply:
Sample Creator must also stop if the system is already at max number of samples (can't have this unbounded as we'll run out of DRAM space)

Reply:
Yeah, I agree. But how would you decide on max samples?